lib/workload/components/python-filemanager-tools-layer/filemanager_tools_layer/src/filemanager_tools/utils/file_helpers.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+from functools import reduce
+from operator import concat
+from typing import List, Dict
+
+from .errors import S3FileNotFoundError, S3DuplicateFileCopyError
+from .models import FileObject
+from .aws_helpers import get_bucket_key_pair_from_uri
+from .request_helpers import get_request_response_results, get_response
+from .globals import S3_LIST_ENDPOINT
+from datetime import datetime, timedelta, timezone
+from urllib.parse import urlparse
+from itertools import batched
+
+
+def get_file_object_from_s3_uri(s3_uri: str) -> FileObject:
+    s3_bucket, s3_key = get_bucket_key_pair_from_uri(s3_uri)
+
+    response = get_request_response_results(S3_LIST_ENDPOINT, {
+        "bucket": s3_bucket,
+        "key": s3_key,
+        "currentState": 'true'
+    })
+
+    if len(response) == 0:
+        # Try again with current_state=False
+        response = get_request_response_results(S3_LIST_ENDPOINT, {
+            "bucket": s3_bucket,
+            "key": s3_key,
+            "currentState": 'false'
+        })
```

Comment:
Non-current records might not have any data on S3, except for something like old versioned objects. Just something to keep in mind, I'm not 100% sure if this is being used to presign URLs eventually.